syntax = "proto3";

option csharp_namespace = "digma_collector";

package collector;

// The greeting service definition.
service DigmaCollector {
  // Sends a greeting
  rpc Export (ExportRequest) returns (ExportResponse);
}

// The request message containing the user's name.
message ExportRequest {
  string span_id =1;
  string trace_id =2;
  string service_name =3;

  // The request message containing the user's name.
  
  message ErrorInformation {
    string exception_name = 1;
    string exception_type = 2;
    string exception_stack = 3;
    string timestamp =4;
  }

  repeated ErrorInformation errors =4;
}

// The response message containing the greetings.
message ExportResponse {
  string message = 1;
}